core/src/test/java/org/jboss/pnc/build/finder/pnc/client/HashMapCachingClientTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.build.finder.pnc.client;
+
+import org.jboss.pnc.client.RemoteCollection;
+import org.jboss.pnc.client.RemoteResourceException;
+import org.jboss.pnc.dto.Artifact;
+import org.jboss.pnc.dto.BuildPushResult;
+import org.jboss.pnc.dto.ProductVersion;
+import org.junit.Assert;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * @author Jakub Bartecek
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class HashMapCachingClientTest {
+
+    private HashMapCachingPncClient cachingPncClient;
+
+    private DummyPncClient dummyPncClient;
+
+    {
```

Comment:
It is kind of strange to initialize here, no? I guess the only alternative is to make them static? JUnit has a `@BeforeClass` but it must be a static method.

Reply:
I think, it doesn't matter in this case as it is a test class, which should have only one instance anyway. I did it this way for some reason, which I cannot recall right now :-D